rs-matter-macros-impl/src/idl.rs

```diff
@@ -179,14 +179,16 @@ fn field_type(f: &DataType) -> TokenStream {
         "namespace" => quote!(u8),
         "tag" => quote!(u8),
 
+        // Items with lifetime. If updating this, remember to add things to
+        // [needs_lifetime]
+        "char_string" => quote!(#krate::tlv::UtfStr<'a>),
+        "long_char_string" => quote!(#krate::tlv::UtfStr<'a>),
+        "octet_string" => quote!(#krate::tlv::OctetStr<'a>),
+        "long_octet_string" => quote!(#krate::tlv::OctetStr<'a>),
+
         // Several structs and unsupported bits.
         // TODO: at least strings should be supported
-        //       strings should be UtfStr
-        //       octet_string should be OctetStr
-        //
-        // However if we use strings, we should propagate a lifetime ('a ?) to top level
-        "char_string" | "long_char_string" | "long_octet_string" | "octet_string" | "ipadr"
-        | "ipv4adr" | "ipv6adr" | "ipv6pre" | "hwadr" | "semtag" | "tod" | "date" => {
+        "ipadr" | "ipv4adr" | "ipv6adr" | "ipv6pre" | "hwadr" | "semtag" | "tod" | "date" => {
```

Comment:
Just mentioning, I do realize this scope is not part of this PR: I think supporting all of `ipadr`, `ipv4adr`, `ipv6adr` is not a problem anymore, because recently the corresponding Rust types were moved to the `core::net` module.

As for `hwadr`, maybe `type HwAddr = [u8; 6]` (that is, if Matter's hwadr is exactly 6 bytes)?

BTW: Speaking of interesting abbreviations... this is the first time I see `Address` abbreviated to `Adr` with a **single** "d"...

Reply:
Update. Well, [this](https://www.allacronyms.com/address/abbreviated) things otherwise. Oh well...

Reply:
I would like to incrementally add these when we hit them. I believe for a first pass probably just having `[u8; <size>]` is correct however I wanted to double-check usages. Opaque byte arrays are annoying to work with so we may want to add wrappers instead ... we will have to see.

Reply:
+1 about "adr" being odd, however:

![image](https://github.com/project-chip/rs-matter/assets/1832280/9aeeee79-2f50-4a92-9192-ddc5ed9f4ce2)


spec seems to have it and then it moved around. I think it is now like this to stay.